cmd/migrate.go

```diff
@@ -0,0 +1,287 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	config "github.com/speakeasy-api/sdk-gen-config"
+	"github.com/speakeasy-api/sdk-gen-config/workflow"
+	"github.com/speakeasy-api/speakeasy/internal/model"
+	"gopkg.in/yaml.v3"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const defaultSourceName = "my-source"
+
+type MigrateFlags struct {
+	Directory   string `json:"directory"`
+	GenFilename string `json:"gen-filename"`
+}
+
+var migrateCmd = &model.ExecutableCommand[MigrateFlags]{
+	Usage:  "migrate",
+	Short:  "migrate to v15 of the speakeasy workflow + action",
+	Long:   "migrate to v15 of the speakeasy workflow + action",
+	Hidden: true,
+	Run:    migrateFunc,
+	Flags: []model.Flag{
+		model.StringFlag{
+			Name:         "directory",
+			Shorthand:    "d",
+			Description:  "directory to migrate. Expected to contain a `.github/workflows` directory. Defaults to `.`",
+			DefaultValue: ".",
+		},
+		model.StringFlag{
+			Name:         "gen-filename",
+			Shorthand:    "g",
+			Description:  "name of the file containing the generation workflow. Defaults to `speakeasy_sdk_generation.yml`",
```

Comment:
Unrelated to this PR this is correct but just a meta note. I'm making the default name in the new github workflows speakeasy_sdk_generation.yaml.

We use .yaml extensions for workflows, gen, gen-lock. Mine as well make everything consistent.

Reply:
Sounds good